src/sessions/video-session-handler.ts

```diff
@@ -216,33 +218,58 @@ export class VideoSessionHandler extends BaseSessionHandler {
   }
 
   // triggers a propose from the backend
-  async startSession (startParams: IStartVideoSessionParams): Promise<{ conversationId: string }> {
-    let participant: { address: string };
+  async startSession(startParams: IStartVideoSessionParams | IStartVideoMeetingSessionParams): Promise<{ conversationId: string }> {
+    if ((<IStartVideoSessionParams>startParams).jid) {
+      const conferenceParams = (<IStartVideoSessionParams>startParams);
+      let participant: { address: string };
 
-    if (startParams.inviteeJid) {
-      participant = { address: startParams.inviteeJid };
-    } else {
-      participant = { address: this.sdk._personDetails.chat.jabberId };
-    }
+      if (conferenceParams.inviteeJid) {
+        participant = { address: conferenceParams.inviteeJid };
+      } else {
+        participant = { address: this.sdk._personDetails.chat.jabberId };
+      }
 
-    const data = JSON.stringify({
-      roomId: startParams.jid,
-      participant
-    });
+      const data = JSON.stringify({
+        roomId: conferenceParams.jid,
+        participant
+      });
 
-    this.requestedSessions[startParams.jid] = true;
+      this.requestedSessions[conferenceParams.jid] = true;
 
-    try {
-      const response = await requestApi.call(this.sdk, `/conversations/videos`, {
-        method: 'post',
-        data
+      try {
+        const response = await requestApi.call(this.sdk, `/conversations/videos`, {
+          method: 'post',
+          data
+        });
+
+        return { conversationId: response.data.conversationId };
+      } catch (err) {
+        delete this.requestedSessions[conferenceParams.jid];
+        this.log('error', 'Failed to request video session', err);
+        throw err;
+      }
+    } else {
+      const meetingParams = (<IStartVideoMeetingSessionParams>startParams);
+      const participant = { address: this.sdk._personDetails.chat.jabberId };
+      const data = JSON.stringify({
+        meetingId: meetingParams.meetingId,
+        participant
       });
 
-      return { conversationId: response.data.conversationId };
-    } catch (err) {
-      delete this.requestedSessions[startParams.jid];
-      this.log('error', 'Failed to request video session', err);
-      throw err;
+      this.requestedMeetingSessions[meetingParams.meetingId] = true;
+
+      try {
+        const response = await requestApi.call(this.sdk, `/conversations/videos/participants`, {
+          method: 'post',
+          data
+        });
+
+        return { conversationId: response.data.conversationId };
+      } catch (err) {
+        delete this.requestedMeetingSessions[meetingParams.meetingId];
+        this.log('error', 'Failed to request video session', err);
```

Comment:
We should distinguish this error from the other one.

Reply:
Good point. How about `Failed to request video session for conference ID/meeting ID`?